platform/macos/display_server_macos.h

```diff
@@ -64,7 +64,7 @@
 #undef CursorShape
 
 class DisplayServerMacOS : public DisplayServer {
-	// No need to register with GDCLASS, it's platform-specific and nothing is added.
+	GDCLASS(DisplayServerMacOS, DisplayServer); // Note: required for Object::cast_to.
```

Comment:
> Note: required for Object::cast_to.

This is currently false, but will be required after https://github.com/godotengine/godot/pull/103708